index.js

```diff
@@ -8,6 +8,13 @@ var util = require("util");
 var exec = require("child_process").exec;
 
 function PDFImage(pdfFilePath, options) {
+  // validating the file path for invalid characters to prevent remote code execution
+  var filter_chars = /[`!@#$%^&*()_+\-=\[\]{};':"\\|,<>\/?~]/;
```

Comment:
Not all characters could trigger remote code execution.
You've blacklisted some harmful and probably legit one such as _ (underscore), ~ (tilde), etc.

I would only filter these characters, that have potential command shell feature:
;|`$()&<>

Reply:
Oh yeah didn't think about it. I did it in a hurry and was sleepy! 😅